lib/media/ttml_text_parser.js

```diff
@@ -217,6 +208,25 @@ shaka.media.TtmlTextParser.getLeafNodes_ = function(element) {
 
 
 /**
+ * Insert \n where <br> tags are found
+ *
+ * @param {!Node} element
```

Comment:
Try iterating over .children instead of .childNodes
(The credit for the idea goes to @TheModMaker :))